### ISSUE TITLE
Make state a dropdown?

### ISSUE THREAD

**Issue by normon66** [mplp/docassemble-mlhframework#90]
For consistency and to prevent spelling mistakes or incorrect state abbreviations, I feel that we should make state fields dropdowns. 

![image](https://github.com/mplp/docassemble-mlhframework/assets/32527754/82f0df00-387f-47c6-b423-45a15d1fd5c7)

**Comment by ekressmiller** [comment]
I think @BryceStevenWilley concluded we could not do this because of difficulties getting accurate lists (especially internationally) and because of the way it works with the Google-maps integration. I will see if I can go back and find the notes on that.

**Comment by BryceStevenWilley** [comment]
That is correct: https://github.com/SuffolkLITLab/docassemble-AssemblyLine/issues/281 has the most details, but https://github.com/mplp/docassemble-mlhframework/pull/36 and https://github.com/SuffolkLITLab/docassemble-AssemblyLine/pull/794 are also relevant.

Specifically, if we only did US addresses, Google Autocomplete worked fine with a state drop down. However, if we added in international addresses, Google autocomplete wouldn't fill in the state correctly, due to the state drop down not getting populated with the countries states correctly before the autocomplete occurs. I was also a little skeptical about the best way to fill in states in international mailing address, and that maybe the state list we were using wouldn't be the best option there.

I was okay with the spelling mistakes that you mentioned because Google Autocomplete would fill in the correct state by the time folks get to the city field in > 95% of the time in my estimate. We could revisit if we have evidence that folks are still spelling things incorrectly there, but I'm not a huge fan having a separate question for international addresses.
